### PR TITLE
Big version update

### PR DESCRIPTION
**Pull request by eranrund**
### Motivation

As ToB pointed out, some of our dependencies are not up to date. This is the first take at upreving a bunch of our deps, focusing on RustCrypto-related ones.

### In this PR
* Upreving of a bunch of deps

### Future Work
* Unfortunately, for this to work, I had to fork `schnorrkel` (https://github.com/eranrund/schnorrkel/commit/9fe3a3c840f3eaa348f55428c40f150bf09aa608), `ed25519-dalek` (https://github.com/eranrund/ed25519-dalek/commit/ffb1db5edf1fcb5aaebfb3a54dbba98072dbafbb) and `bulletproofs` (https://github.com/eranrund/bulletproofs/commit/e8e8ef45ecc6d31f1a9525140edc977351d0f780). These uprevs may or may not be tricky to get in:
   - `schnorrkel` depends on `ed25519-dalek`, so until that goes in there's no chance.
   - `ed25519-dalek` is pretty unobtrusive but it does require a new major version due to moving to a newer `curve25519-dalek` which updated its major version (`Update the digest dependency to 0.9. This requires a major version because the digest traits are part of the public API, but there are otherwise no changes to the API.`)
  - `bulletproofs` - This one required a newer rust toolchain, so I am not sure how they're going to feel about that. I need to see why that was needed.

----

I'm putting this draft here to see what people think. I still need to see it successfully deployed, but I'm getting closer to achieving that.

### CODE REVIEW

**Comment by eranrund** [comment]
This passed deployment, but now needs a hack around https://github.com/RustCrypto/utils/blob/master/cpuid-bool/src/lib.rs#L96 because that doesn't work for how we build enclaves... TBD what can be done about that.

File: crypto/keys/Cargo.toml [hunk 1]
```diff
@@ -17,21 +17,18 @@ mc-util-repr-bytes = { path = "../../util/repr-bytes" }
 mc-util-serial = { path = "../../util/serial", default-features = false }
 prost = { version = "0.6.1", default-features = false, features = ["prost-derive"] }
 rand_core = { version = "0.5", default-features = false }
-# Note: Signature must be pinned because signature 1.2 depends on digest = 0.9, but
-# ed25519-dalek gets digest from curve25519-dalek re-export, and curve25519-dalek is on digest 0.8,
-# we cannot move forward with digest 0.9 until curve25519-dalek crate does
-signature = { version = "=1.0", default-features = false, features = ["digest-preview"] }
```
Comment (garbageslam) [inline crypto/keys/Cargo.toml:23]: :tada:

**Comment by eranrund** [comment]
Slam a tiny better better than `master`. This is `master`:
image

This is this branch:
image

My next steps would be to rebase this on top of `master` to resolve the merge conflicts, and to switch back to the official `schnorrkel` repo that has accepted my PR but has not yet released a new version to crates.io.

**Comment by eranrund** [comment]
Note that this only focuses on RustCrypto deps (and stuff that depends on their crates). A separate PR needs to tackle other updates.

**Comment by eranrund** [comment]
Slammed again, better throughput but worse completion times:
image

**Comment by eranrund** [comment]
And another slam, better than the previous two:
image

**Pull request merged**